app/lib/actions/katello/repository/finish_upload.rb

```diff
@@ -5,13 +5,22 @@ class FinishUpload < Actions::Base
         def plan(repository, options = {})
           dependency = options.fetch(:dependency, nil)
           generate_metadata = options.fetch(:generate_metadata, true)
+          sync_capsule = options.fetch(:sync_capsule, true)
           plan_action(Katello::Repository::MetadataGenerate, repository, :dependency => dependency) if generate_metadata
 
           recent_range = 5.minutes.ago.utc.iso8601
           plan_action(Katello::Repository::FilteredIndexContent,
                       id: repository.id,
                       filter: {:association => {:created => {"$gt" => recent_range}}},
                       dependency: dependency)
+          plan_self(sync_capsule: sync_capsule, repository_id: repository.id)
+        end
+
+        def run
+          if input[:sync_capsule]
```

Comment:
This failed for me because the current user wasn't set, the resulting action errors witH:
`No current user is set. Please set User.current to perform a remote action`

take a look at the KeepCurrentUser Middleware